src/PickerAvoidingView/index.ios.js

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import { View } from 'react-native';
+import { PickerStateContext } from "../PickerStateProvider";
+import { IOS_MODAL_HEIGHT } from "../constants";
+
+const styles = {
+    container: {
+        flex: 1,
+        flexDirection: 'column',
+        justifyContent: 'flex-start',
+        alignItems: 'stretch',
+    },
+    top: {
+        flex: 1,
+    },
+    bottom: {
+        height: IOS_MODAL_HEIGHT,
+    },
+};
+
+/**
+ * PickerAvoidingView is a React component that adjusts the view layout to avoid
+ * being covered by an open iOS UIPickerView modal. It's meant to be similar to
+ * the built-in KeyboardAvoidingView component, but specifically tailored for
+ * iOS picker modals.
+ *
+ * In order for this component to work correctly, all the pickers and the
+ * PickerAvoidingView should have a PickerStateProvider ancestor.
+ *
+ * @param {React.ReactNode} props.children - The child components that should be
+ * protected from obstruction by the picker modal
+ */
+export function PickerAvoidingView(props) {
+    const context = React.useContext(PickerStateContext);
+    const isPickerOpen = context && context.isPickerOpen;
+
+    return (
+        <View style={styles.container}>
+            <View style={styles.top}>
```

Comment:
Maybe let's add this when it's needed? As always, it can potentially break the solution, and we don't need it currently. If it will be requested in the upstream, it's another topic.

Reply:
Just an intuition that it's safer to control the flexbox as a whole here, both the container and the items. In theory, the injected child/children could have some CSS style for flex items, which I didn't really want to support.

Reply:
Thanks for the additional context. Maybe we could add a short comment explaining the reasoning behind this to make it obvious to others?